src/Transport/Sending/MessageDispatcher.cs

```diff
@@ -59,8 +58,6 @@ public async Task Dispatch(TransportOperations outgoingMessages, TransportTransa
                         if (!defaultOperationsPerDestination.ContainsKey(destination))
                         {
                             defaultOperationsPerDestination[destination] = [operation];
-                            // because we batch only the number of destinations are relevant
```

Comment:
Getting rid of this here also reduces the cognitive overhead because this code doesn't have knowledge anymore about the inner workings of the batching path.